.github/workflows/tests.yml

```
@@ -0,0 +1,57 @@
+name: tests
+
+on:
+  #    push:
+  pull_request:
+  #  schedule:
+  #  - cron: '0 0 * * *'
+
+jobs:
+  tests:
+    strategy:
+      matrix:
+        os: [macOS, Ubuntu, Windows]
+        php: [7.0, 7.1, 7.2]
+
+        include:
+          - os: macOS
+            experimental: false
+            os-version: macos-latest
+
+          - os: Ubuntu
+            experimental: true
+            os-version: ubuntu-latest
+
+          - os: Windows
+            experimental: true
+            os-version: windows-latest
+
+    name: ${{ matrix.os }} - PHP ${{ matrix.php }} --${{ matrix.dependency }}
+
+    continue-on-error: ${{ matrix.experimental }}
```

We can drop the experimental concept on this one. 
```suggestion
        include:
          - os: macOS
            os-version: macos-latest

          - os: Ubuntu
            os-version: ubuntu-latest

          - os: Windows
            os-version: windows-latest

    name: ${{ matrix.os }} - PHP ${{ matrix.php }}
```